pkg/githubtemplate/github_template.go

```diff
@@ -10,8 +10,19 @@ import (
 	"gopkg.in/yaml.v3"
 )
 
-// FindNonLegacy returns the list of template file paths from the template folder (according to the "upgraded multiple template builder")
-func FindNonLegacy(rootDir string, name string) []string {
+// TemplateHandler contains all the template related functions
+type TemplateHandler struct {
+	// FindNonLegacy returns the list of template file paths from the template folder (according to the "upgraded multiple template builder")
+	FindNonLegacy func(rootDir string, name string) []string
+	// FindLegacy returns the file path of the default(legacy) template
+	FindLegacy func(rootDir string, name string) *string
+	// ExtractName returns the name of the template from YAML front-matter
+	ExtractName func(filePath string) string
+	// ExtractContents returns the template contents without the YAML front-matter
+	ExtractContents func(filePath string) []byte
+}
+
+func findNonLegacy(rootDir string, name string) []string {
```

Comment:
I unexported these, so that we don't use them directly by mistake, and are reminded to use the handler.